addons/binding/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/internal/ICloudTlsCertificateProvider.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.icloud.internal;
+
+import java.net.URL;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.io.net.http.TlsCertificateProvider;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * Provides a TrustManager for https://fmipmobile.icloud.com
+ *
+ * @author Martin van Wingerden - Initial Contribution
+ */
+@Component
+@NonNullByDefault
+public class ICloudTlsCertificateProvider implements TlsCertificateProvider {
+
+    @Override
+    public String getHostName() {
+        return "fmipmobile.icloud.com";
+    }
+
+    @Override
+    public URL getCertificate() {
+        URL resource = Thread.currentThread().getContextClassLoader().getResource("fmipmobile.crt");
+        if (resource != null) {
+            return resource;
+        } else {
+            throw new IllegalStateException("Certifcate resource not found or not accesible");
```

Comment:
I had started typing exactly this suggestion, but didn't finish to submit it earlier today.
But it is nice to see that you came up with exactly the same solution :-)

Did you check whether the framework is robust against RuntimeExceptions that are thrown from here?

Reply:
Yes works great! (I might say surprisingly well )

> java.lang.IllegalStateException: Certifcate resource not found or not accesible
	at org.openhab.binding.icloud.internal.ICloudTlsCertificateProvider.getCertificate(ICloudTlsCertificateProvider.java:37)
	at org.eclipse.smarthome.io.net.http.internal.TlsCertificateTrustManagerAdapter.<init>(TlsCertificateTrustManagerAdapter.java:44)
	at org.eclipse.smarthome.io.net.http.internal.ExtensibleTrustManager.addTlsCertificateProvider(ExtensibleTrustManager.java:162)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)

> 018-11-08 18:49:47.955 [TRACE] [i.n.h.i.ExtensibleTrustManager:114  ] - No specific trust manager found, falling back to default
2018-11-08 18:49:48.030 [WARN ] [i.h.ICloudAccountBridgeHandler:86   ] - Unable to refresh device datajava.io.IOException: java.util.concurrent.ExecutionException: javax.net.ssl.SSLHandshakeException: General SSLEngine problem
	at org.eclipse.smarthome.io.net.http.HttpUtil.executeUrlAndGetReponse(HttpUtil.java:259)
	at org.eclipse.smarthome.io.net.http.HttpUtil.executeUrl(HttpUtil.java:156)
	at org.eclipse.smarthome.io.net.http.HttpUtil.executeUrl(HttpUtil.java:131)
	at org.eclipse.smarthome.io.net.http.HttpRequestBuilder.getContentAsString(HttpRequestBuilder.java:135)
	at org.openhab.binding.icloud.internal.ICloudConnection.callApi(ICloudConnection.java:83)
	at org.openhab.binding.icloud.internal.ICloudConnection.requestDeviceStatusJSON(ICloudConnection.java:64)
	at org.openhab.binding.icloud.handler.ICloudAccountBridgeHandler.lambda$0(ICloudAccountBridgeHandler.java:84)
	at org.eclipse.smarthome.core.cache.ExpiringCache.refreshValue(ExpiringCache.java:81)
	at org.eclipse.smarthome.core.cache.ExpiringCache.getValue(ExpiringCache.java:61)
	at org.openhab.binding.icloud.handler.ICloudAccountBridgeHandler.refreshData(ICloudAccountBridgeHandler.java:135)
	at org.openhab.binding.icloud.handler.ICloudAccountBridgeHandler.lambda$1(ICloudAccountBridgeHandler.java:126)
Caused by: java.util.concurrent.ExecutionException: javax.net.ssl.SSLHandshakeException: General SSLEngine problem
	at org.eclipse.jetty.client.util.FutureResponseListener.getResult(FutureResponseListener.java:118)
	at org.eclipse.jetty.client.util.FutureResponseListener.get(FutureResponseListener.java:110)
	at org.eclipse.jetty.client.HttpRequest.send(HttpRequest.java:676)
	at org.eclipse.smarthome.io.net.http.HttpUtil.executeUrlAndGetReponse(HttpUtil.java:250)
	at org.eclipse.smarthome.io.net.http.HttpUtil.executeUrl(HttpUtil.java:156)
	at org.eclipse.smarthome.io.net.http.HttpUtil.executeUrl(HttpUtil.java:131)
	at org.eclipse.smarthome.io.net.http.HttpRequestBuilder.getContentAsString(HttpRequestBuilder.java:135)
	at org.openhab.binding.icloud.internal.ICloudConnection.callApi(ICloudConnection.java:83)
	at org.openhab.binding.icloud.internal.ICloudConnection.requestDeviceStatusJSON(ICloudConnection.java:64)
	at org.openhab.binding.icloud.handler.ICloudAccountBridgeHandler.lambda$0(ICloudAccountBridgeHandler.java:84)
	at org.eclipse.smarthome.core.cache.ExpiringCache.refreshValue(ExpiringCache.java:81)
Caused by: javax.net.ssl.SSLHandshakeException: General SSLEngine problem